src/aliceVision/lensCorrectionProfile/lcp.hpp

```diff
@@ -0,0 +1,733 @@
+     * @brief Indicate that chromatic models are avaialble
+     * @return true if chromatic models are avaialble
+     */
+    bool hasChromaticParams() const { return _hasChromaticParams; }
+
+    /**
+    * @brief Set fisheye status
+    * @param[in] fisheye status
+    */
+    void setFisheyeStatus(bool s) { _isFisheye = s; }
+
+    /**
+    * @brief Set vignetting availabilty status
+    * @param[in] vignetting availabilty status
+    */
+    void setVignetteParamsStatus(bool s) { _hasVignetteParams = s; }
+
+    /**
+    * @brief Set chromatic models availabilty status
+    * @param[in] chromatic models availabilty status
+    */
+    void setChromaticParamsStatus(bool s) { _hasChromaticParams = s; }
+
+    /**
+    * @brief Chromatic Green model parameters
+    */
+    RectilinearModel ChromaticGreenParams;
+    /**
+    * @brief Chromatic Red/Green model parameters
+    */
+    RectilinearModel ChromaticRedGreenParams;
+    /**
+    * @brief Chromatic Blue/Green model parameters
+    */
+    RectilinearModel ChromaticBlueGreenParams;
+
+    /**
+    * @brief Fisheye model parameters
+    */
+    FisheyeModel fisheyeParams;
+    /**
+    * @brief Pinhole model parameters
+    */
+    //PerspectiveModel perspParams;
+    RectilinearModel perspParams;
+    /**
+    * @brief Vignetting model parameters
+    */
+    VignetteModel vignParams;
+
+    /**
+    * @brief Camera settings
+    */
+    settingsInfo camData;
+
+private:
+
+    bool _isFisheye = false;
+    bool _hasVignetteParams = false;
+    bool _hasChromaticParams = false;
+};
+
+/**
+ * @brief LCPinfo loads and hosts the content of a Lens Correction Profile (LCP) file,
+ * parameters of distortion, vignetting and chromatic aberation models for different
+ * set of camera settings (focal length, focus distance, aperture value).
+ * Detailed information on LCP file content can be found in the Adobe technical report
+ * "Adobe Camera Model" part of the documentation of the Adobe free tool Lens Profile Creator.
+ */
+class LCPinfo
+{
+public:
+    LCPinfo() = default;
+
+    /**
+    * @brief LCPinfo constructor
+    * @param[in] filename The lcp path on disk
+    * @param[in] fullParsing Load only common camera and lens info  and skip all models when set to false (default = true)
+    */
+    LCPinfo(const std::string& filename, bool fullParsing=true);
+    ~LCPinfo() = default;
+
+    /**
+    * @brief LCPinfo loader
+    * @param[in] filename The lcp path on disk
+    * @param[in] fullParsing Load only common camera and lens info  and skip all models when set to false (default = true)
+    */
+    void load(const std::string& filename, bool fullParsing = true);
+
+    /**
+    * @brief Get distortion parameters for a given couple focal length, focus distance. Focus distance can set to zero.
+    * @param[in] focalLength Focal length in mm
+    * @param[in] focusDistance Focus distance in meters
+    * @param[in] lparam Lens parameters to be populated with the distortion model 
```

Comment:
`param[out]` instead of `param[in]` since it is meant to be populated ?

Reply:
Done